sdk/ai/ai-inference-rest/samples/v1-beta/javascript/telemetry.js

```diff
@@ -0,0 +1,174 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/**
+ * Demonstrates how to get instrumentation by open telemetry.
+ *
+ * @summary get instrumentation by open telemetry.
+ */
+
+const { diag, DiagConsoleLogger, DiagLogLevel } = require("@opentelemetry/api");
+const { registerInstrumentations } = require("@opentelemetry/instrumentation");
+const { createAzureSdkInstrumentation } = require("@azure/opentelemetry-instrumentation-azure-sdk");
+const { ConsoleSpanExporter, NodeTracerProvider, SimpleSpanProcessor } = require("@opentelemetry/sdk-trace-node");
+
+diag.setLogger(new DiagConsoleLogger(), DiagLogLevel.VERBOSE);
+
+const provider = new NodeTracerProvider();
+provider.addSpanProcessor(new SimpleSpanProcessor(new ConsoleSpanExporter()));
+provider.register();
+
+// register Azure SDK Instrumentation.
+// ************** IMPORTANT SETUP STEP ***************
+// this must be done before loading @azure/core-tracing
+registerInstrumentations({
+  instrumentations: [createAzureSdkInstrumentation()],
+});
+
+
+const ModelClient = require("@azure-rest/ai-inference").default;
+const { isUnexpected } = require("@azure-rest/ai-inference");
+const { AzureKeyCredential } = require("@azure/core-auth");
+const { createTracingClient } = require("@azure/core-tracing");
+
+const tracingClient = createTracingClient({
+  namespace: "Microsoft.OtelSample",
+  packageName: "otel-sample",
+  packageVersion: "1.0.0",
+});
+
+// Load the .env file if it exists
+require("dotenv").config();
+
+// You will need to set these environment variables or edit the following values
+const endpoint = process.env["ENDPOINT"] || "<endpoint>";
+const key = process.env["KEY"] || "<key>";
+
+async function main() {
+  console.log("== Chat Completions Sample ==");
+
+  // initialize a span named "main" with default options for the spans
+  await tracingClient.withSpan("main", {}, async (updatedOptions) => {
```

Comment:
Please use OTel API here - this is end-user code - 
```suggestion
  await tracer.startActiveSpan('rollTheDice', (span: Span) => {
```

https://opentelemetry.io/docs/languages/js/instrumentation/#create-spans